### ISSUE TITLE
Postgres: LANGUAGE option of CREATE FUNCTION does not supported quoted languages

### ISSUE THREAD

**Issue by enote-kane** [sqlfluff/sqlfluff#3056]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing issues for function declarations.

### Expected Behaviour

Rule errors, if any.

### Observed Behaviour

```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE OR
                       | REPLACE FUNCTION unaccent_immu...'
L:   9 | P:   1 |  PRS | Line 9, Position 1: Found unparsable section: 'CREATE OR
                       | REPLACE FUNCTION unaccent_immu...'
L:  19 | P:   1 |  PRS | Line 19, Position 1: Found unparsable section: 'CREATE
                       | OR REPLACE FUNCTION unify_string(...'
```

### How to reproduce

```sql
CREATE OR REPLACE FUNCTION unaccent_immutable(REGDICTIONARY, TEXT)
    RETURNS TEXT
    LANGUAGE 'c'
    IMMUTABLE
    STRICT
    PARALLEL SAFE
AS '$libdir/unaccent', 'unaccent_dict';

CREATE OR REPLACE FUNCTION unaccent_immutable(TEXT)
    RETURNS TEXT
    LANGUAGE 'sql'
    IMMUTABLE
    STRICT
    PARALLEL SAFE
AS $function$
SELECT unaccent_immutable(REGDICTIONARY 'public.unaccent', $1);
$function$;

CREATE OR REPLACE FUNCTION unify_string(text_ TEXT)
    RETURNS TEXT
    LANGUAGE 'plpgsql'
    IMMUTABLE
    STRICT
    PARALLEL SAFE
AS $function$
BEGIN
    RETURN regexp_replace(unaccent_immutable(lower(text_)), '[^[:alnum:]]', '', 'g');
END
$function$;
```

### Dialect

postgres

### Version

sqlfluff, version 0.12.0

### Configuration

```
[sqlfluff]
nocolor = True
dialect = postgres
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
The issue is the `LANGUAGE 'c'` parameter. Changing it to `LANGUAGE c` allows it to parse.

The [Postgres Create Function documentation](https://www.postgresql.org/docs/current/sql-createfunction.html#:~:text=Enclosing%20the%20name%20in%20single%20quotes%20is%20deprecated%20and%20requires%20matching%20case.) has this to say:

> Enclosing the name in single quotes is deprecated and requires matching case.

So technically it's valid postgres, and would be easy enough to add to SQLFluff, but given it's deprecated, not sure we should add it. WDYT?

**Comment by WittierDinosaur** [comment]
I think with this one we should add it. If it's deprecated but still compiles in Postgres then it should in SQLFluff (we could even have a dialect-specific rule that this fails).

**Comment by tunetheweb** [comment]
> we could even have a dialect-specific rule that this fails

Knew that would be coming next 😁

Do think we should try to avoid adding a rule for every individual dialect deprecation. As we know from the other conversations rules aren’t free and have four “costs”:
1. Initial development.
2. Performance costs of lint (and CI!).
3. More confusing config.
4. Ongoing maintenance.
5. 

Think we’ve got to judge the value of a rule versus those costs. For commonly used, but bad practice/deprecated, it makes sense. For others (like this one?) maybe the unparsable error serves as enough as it’s easily fixed?

But then I’m not sure how common this is? Or if I’m under-estimating how easily it is to fix. Interested to hear @enote-kane ’s opinion on this as the original raiser.

**Comment by enote-kane** [comment]
@tunetheweb @WittierDinosaur TBH, I would like to get this reported as using a deprecated feature, but it certainly should parse it, at least.

Having a whole function not being parsed due to this is very unfortunate for me, since I have lot's of functions and and also pretty complex ones.

Nevertheless, I now at least know why it is not getting parsed. So thanx for that info.

**Comment by tunetheweb** [comment]
The parsing is fixed in #3068
